### ISSUE TITLE
[Date Picker, Range Date Picker]: [X] remove date button is not accessible from the keyboard

### ISSUE THREAD

**Issue by NatalliaAlieva** [epam/UUI#2719]
## Description
[X] (remove the date) button in the Date Picker and Rage Date Picker is not accessible from the keyboard.

## Steps to Reproduce
1. Open [Date Picker](https://uui.epam.com/documents?id=datePicker&mode=doc&category=components) OR [Range Date Picker](https://uui.epam.com/documents?id=rangeDatePicker&mode=doc&category=components) component.
2. Select any dates.
3. Use `Tab` key to navigate to the field(s) with selected date(s).
4. Use `Tab` key to navigate to the [X] button.

## Actual result
[X] button is not accessible from the keyboard.

https://github.com/user-attachments/assets/0ec4d67c-c2d0-476c-bb73-05b0d98089e3

## Expected result
[X] (remove date) button in in the Date Picker and Rage Date Picker should be accessible from the keyboard, should have visible focus indicator and should have accessible name, e.g., Remove date.

**Failed WCAG Checkpoint:** 2.1.1 Keyboard (Level A)     
https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html

**Comment by NatalliaAlieva** [comment]
1. Range Date Picker — [Clear Button Issue]

https://github.com/user-attachments/assets/ff37c47a-8f2a-485f-9a65-0986f6fce781

2. Picker Input — [Icon Click Issue], [Clear Button Issue]

https://github.com/user-attachments/assets/a65f1091-33c5-4b30-8938-987a2690aff5

3. Search Input — [Icon Click Issue], [Clear Button Issue]

https://github.com/user-attachments/assets/af145598-17e1-4e42-8b8f-880b6249bc82

4. Tag — [Icon Click Issue], [Clear Button Issue]

https://github.com/user-attachments/assets/2353c994-a10e-4547-b240-768c694dcd33

5. Text Input — [Icon Click Issue], [Clear Button Issue]

https://github.com/user-attachments/assets/ee1af5f8-a039-4296-8b8c-c1f03f125544

6. Time Picker — [Clear Button Issue]

https://github.com/user-attachments/assets/92e406f7-dc8d-4e73-943a-70b6c02276cd
